Service/TransformContentTypeService.php

```diff
@@ -69,9 +65,11 @@ public function transform(ContentType $contentType): \Generator
                     }
 
                     $transformer = $this->getTransformer($dataField);
-                    if (!empty($transformer) && $transformer->canTransform(new ContentTransformContext([$dataFieldType]))) {
-                        $dataTransformed = $transformer->transform($data);
-                        if ($transformer->changed($dataTransformed)) {
+                    $contentTransformContext = ContentTransformContext::fromDataFieldType(\get_class($dataFieldType), $data);
+                    if (!empty($transformer) && $transformer->canTransform($contentTransformContext)) {
+                        $dataTransformed = $transformer->transform($contentTransformContext);
+                        $contentTransformContext->setTransformedData($dataTransformed);
+                        if ($transformer->changed($contentTransformContext)) {
```

Comment:
Maybe the transformer does not need this changed function. But only the contentTransformContext? ($context->hasChanges())